apps/aecore/lib/aecore/sync/sync.ex

```diff
@@ -457,11 +463,11 @@ defmodule Aecore.Sync.Sync do
     if sync_in_progress?(peer_id) do
       Logger.info("#{__MODULE__}: Already syncing with #{inspect(peer_id)}")
     else
-      do_start_sync1(peer_id, remote_hash)
+      do_start_sync_internal(peer_id, remote_hash)
     end
   end
 
-  defp do_start_sync1(peer_id, remote_hash) do
+  defp do_start_sync_internal(peer_id, remote_hash) do
```

Comment:
the naming problem is in  `do_start_sync`, as it eventually doesn't start sync. If you fix the naming of this to e.g. `ensure_sync_running` or something that indicates that it would not start again if already running there is no need for `_internal`